apps/designer/app/designer/features/style-panel/shared/css-value-input/unit-select.tsx

```diff
@@ -30,21 +31,31 @@ for (const [key, value] of unitRenderMap.entries()) {
 const defaultUnits = Array.from(unitRenderMap.keys());
 
 type UseUnitSelectType = {
+  property: string;
   value?: UnitValue;
   onChange: (value: StyleValue) => void;
   units?: Array<Unit>;
   onCloseAutoFocus: (event: Event) => void;
 };
 
 export const useUnitSelect = ({
+  property,
   onChange,
   value,
   units = defaultUnits,
   ...props
 }: UseUnitSelectType) => {
   const [isOpen, setIsOpen] = useState(false);
   const renderUnits = useMemo(
-    () => units.map((unit) => unitRenderMap.get(unit) ?? unit),
+    () =>
+      units
+        .filter((unit) => {
+          return isValid(
+            property,
+            value?.value + String(unit).replace("number", "")
```

Comment:
this can use toValue, right?

Reply:
turns out same thing has been already used in `parseCssValue` function which should also use toValue ... we need a single place that turns internal CSS value object to CSS string